platform/darwin/src/NSComparisonPredicate+MGLAdditions.mm

```diff
@@ -140,10 +140,18 @@ - (id)mgl_jsonExpressionObject {
             [NSException raise:NSInvalidArgumentException
                         format:@"NSPredicateOperatorType:%lu is not supported.", (unsigned long)self.predicateOperatorType];
     }
-    if (op) {
-        return @[op, self.leftExpression.mgl_jsonExpressionObject, self.rightExpression.mgl_jsonExpressionObject];
+    if (!op) {
+        return nil;
     }
-    return nil;
+    NSArray *comparisonArray = @[op, self.leftExpression.mgl_jsonExpressionObject, self.rightExpression.mgl_jsonExpressionObject];
+    if (self.options) {
+        NSDictionary *collatorObject = @{
+            @"case-sensitive": @(!(self.options & NSCaseInsensitivePredicateOption)),
+            @"diacritic-sensitive": @(!(self.options & NSDiacriticInsensitivePredicateOption)),
+        };
+        return [comparisonArray arrayByAddingObject:@[@"collator", collatorObject]];
```

Comment:
Does this means `'Québec' =[] 'Quebec'` will use a collator (and thus the current locale) vs. `'Québec' = 'Quebec'` doing non-locale-aware comparison?

Reply:
As long as an option is specified in brackets, it’ll use a collator. To match Core Data behavior, I suppose we could default the `locale` to `en-posix` when `c` or `d` is specified unless `l` is also specified.

A literal `=[]` is invalid syntax; there has to be an option. NSPredicate will raise an exception immediately on initialization.

Reply:
I think the current behavior sounds right. :+1: